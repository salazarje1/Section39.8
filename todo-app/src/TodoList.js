import React, { useState } from 'react'; 
import { v4 as uuid } from "uuid";

import Todo from './Todo';
import NewTodoForm from "./NewTodoForm"; 


const TodoList = () => {
    const INITIALSTATE = [
        {id: uuid(), task: "Walk Dog"},
        {id: uuid(), task: "Make Dinner"}
    ]

    const [tasks, setTasks] = useState(INITIALSTATE); 

    const addTask = (newTask) => {
        console.log(newTask);
        setTasks(tasks => [...tasks, {task: newTask, id: uuid() }]); 
    }

    const removeTask = (id) => {
        setTasks(tasks => tasks.filter(task => task.id !== id)); 
    }

    return (
        <div>
            <h2>Todo List</h2>
            <NewTodoForm addTask={addTask} />
            <div>
                {tasks.map(({ id, task }) => {
                    return <Todo id={id} key={id} task={task} removeTask={removeTask} />
                })}
            </div>
        </div>
    )
}


export default TodoList; 