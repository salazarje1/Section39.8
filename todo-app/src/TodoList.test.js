import { render, fireEvent } from "@testing-library/react";
import TodoList from "./TodoList";

it('should render without crashing', () => {
    render(<TodoList />)
})

it('should match snapshot', () => {
    const { asFragment } = render(<TodoList />);
    expect(asFragment()).toMatchSnapshot(); 
})

it("shoudl add new task", () => {
    const { queryByText, getByLabelText } = render(<TodoList />); 

    const input = getByLabelText("New Task");
    const btn = queryByText('Add New Task'); 

    expect(queryByText("Clean")).not.toBeInTheDocument(); 

    fireEvent.change(input, { target: { value: "Clean"} });
    fireEvent.click(btn);

    expect(queryByText("Clean")).toBeInTheDocument(); 
})