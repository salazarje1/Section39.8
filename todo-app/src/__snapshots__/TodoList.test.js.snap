// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <h2>
      Todo List
    </h2>
    <form>
      <label>
        New Task
        <input
          name="todo"
          placeholder="todo"
          type="text"
          value=""
        />
      </label>
      <button>
        Add New Task
      </button>
    </form>
    <div>
      <div>
        <p>
          Walk Dog
        </p>
        <button>
          Delete
        </button>
      </div>
      <div>
        <p>
          Make Dinner
        </p>
        <button>
          Delete
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
